idioms/str-vs-string.md

```diff
@@ -0,0 +1,89 @@
+# Using &str instead of &String
+
+## Description
+
+It is usually recommended to prefer a `&str` type over a `&String` type for arguments.  In most cases `&str` has more flexibility.  A `&String` will coerce to a `&str` type when required -- however, the opposite is not true.  
```

Comment:
I don't think you need to limit to slice-able or fat pointer types. You should always prefer using the borrowed type over borrowing the owned type, e.g., `&T` to `&Box<T>`